test/new-e2e/utils/infra/stack_manager.go

```diff
@@ -60,27 +60,18 @@ func newStackManager(ctx context.Context) (*StackManager, error) {
 	}, nil
 }
 
-func getNewStack(ctx context.Context, name, project string, deployFunc pulumi.RunFunc, failOnMissing bool) (auto.Stack, error) {
-	var newStack auto.Stack
-	var err error
-
-	if failOnMissing {
-		newStack, err = auto.SelectStackInlineSource(ctx, name, projectName, deployFunc)
-	} else {
-		newStack, err = auto.UpsertStackInlineSource(ctx, name, projectName, deployFunc)
-	}
-	if err != nil {
-		return auto.Stack{}, err
-	}
-
-	return newStack, nil
-}
-
 // GetStack creates or return a stack based on env+stack name
 func (sm *StackManager) GetStack(ctx context.Context, envName string, name string, config auto.ConfigMap, deployFunc pulumi.RunFunc, failOnMissing bool) (auto.UpResult, error) {
 	sm.lock.RLock()
 	defer sm.lock.RUnlock()
 
+	var getStack func(context.Context, string, string, pulumi.RunFunc, ...auto.LocalWorkspaceOption) (auto.Stack, error)
+	if failOnMissing {
+		getStack = auto.SelectStackInlineSource
+	} else {
+		getStack = auto.UpsertStackInlineSource
+	}
+
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
I like this !